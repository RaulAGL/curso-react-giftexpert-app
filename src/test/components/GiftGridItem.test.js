
import { shallow } from 'enzyme';
import GiftGridItem from "../../components/GiftGridItem";
import React from "react";

describe('Probar funcionalidad de giftGridItem',() => {

    const item = {
        title : 'Title',
        url : 'http://test.url'
    };
    const wrapper = shallow(<GiftGridItem {...item} />);


    test('Probar componente GiftGridItem con snapShot', () => {


        expect(wrapper).toMatchSnapshot();
    });


    test('Debe de evaluar el parrafo que contenga el title', () => {

        const p = wrapper.find('p').text().trim();

        expect( p ).toBe( item.title );
    });

    test('Debe de evaluar que imagene contenga src y alt', () => {

        const img = wrapper.find('img');


        expect( img.prop('src') ).toBe( item.url );
        expect( img.prop('alt') ).toBe( item.title );

    });

    test('Debe probar que contenga la clase animate__fadeIn', () => {

        const div = wrapper.find('div');

        expect(div.hasClass('animate__fadeIn')).toBe(true);

    });

});
