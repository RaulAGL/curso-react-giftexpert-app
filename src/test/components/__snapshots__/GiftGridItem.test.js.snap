// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar funcionalidad de giftGridItem Probar componente GiftGridItem con snapShot 1`] = `
<Fragment>
  <div
    className="card animate__animated animate__fadeIn  "
  >
    <img
      alt="Title"
      src="http://test.url"
    />
    <p>
      Title
    </p>
  </div>
</Fragment>
`;
